### ISSUE TITLE
YAJL_GET_DOUBLE(v) fails with German locale

### ISSUE THREAD

**Issue by fsphil** [lloyd/yajl#79]
A German user of my software reported an odd bug where floating point values are being rounded down to a whole number, for example 54.12345 is being returned as 54. I believe the cause is the use of strtod() in yajl_tree.c:

v->u.number.d = strtod(v->u.number.r, &endptr);
https://github.com/lloyd/yajl/blob/master/src/yajl_tree.c#L320

strtod() expects a comma rather than a point when used in the de_DE locale, and probably countless other variations in other locales too. I was able to reproduce this by setting the env variable LC_ALL="de_DE".

**Comment by f2404** [comment]
Yep - came upon it in `package-query` and had to spend some time investigating it.
